main.py

```diff
@@ -854,6 +857,29 @@ def run(
                 break
 
 
+def run_async_update_with_retries(mgr, current_block, logging_path, max_retries=5):
+    failed_async_calls = 0
+
+    while failed_async_calls < max_retries:
+        try:
+            async_update_pools_from_contracts(mgr, current_block, logging_path)
+            return  # Successful execution
+        except AyncUpdateRetryException as e:
+            failed_async_calls += 1
+            remaining_pools = mgr.update_remaining_pools()
+            shuffle(remaining_pools)  # shuffle to avoid repeated immediate failure of the same pool
+            mgr.pools_to_add_from_contracts = remaining_pools
```

Comment:
In conjunction with a previous comment,  function `mgr.update_remaining_pools` can handle this assignment internally

Reply:
fixed